### PR TITLE
16 wall kick

### LINKED ISSUE #16: Wall kicks
Refer to Tetris wiki

### PR DESCRIPTION
**Pull request by EstevanBR**
fix #16

### CODE REVIEW

**Comment by EstevanBR** [comment]
@gillett-hernandez  please review. thx

File: Tetris Clone C++/TetrisPiece.cpp [hunk 1]
```diff
@@ -80,83 +82,151 @@ void TetrisPiece::moveDown() {
...
+        }
+    }
+    if (debug) std::cout << "rotate " << ((rotateFree) ? "free" : "not free") << std::endl;
+    return rotateFree;
+}
+void TetrisPiece::applyRotation(bool newGrid[4][4], int bounds, int colOff, int rowOff) {
+    this->setGrid(newGrid, bounds);
+    this->setShapes();
+    this->updatePosition(colOff, rowOff);
+}
```
Comment (gillett-hernandez) [inline Tetris Clone C++/TetrisPiece.cpp:114]: ambiguous method name, should be update_grid or something

File: Tetris Clone C++/TetrisPiece.cpp [hunk 2]
```diff
@@ -80,83 +82,151 @@ void TetrisPiece::moveDown() {
...
         int bounds = 0;
-        int x = this->gridSize[0];
-        int y = this->gridSize[1];
-        if (x == y) {
-            bounds = x&y;
+        int width = this->gridSize[0];
+        int height = this->gridSize[1];
+        if (width == height) {
+            bounds = width&height;
         }
```
Comment (gillett-hernandez) [inline Tetris Clone C++/TetrisPiece.cpp:125]: if width != height, then bounds = 0???

File: Tetris Clone C++/GridController.cpp [hunk 3]
```diff
@@ -160,13 +156,13 @@ void GridController::printGrid() {
 
 void GridController::saveGridToFile() {
     FILE *fp = fopen(fileName, "wb");
-    int length = ROWS * COLUMNS;
+    size_t length = ROWS * COLUMNS;
     char *gridString = (char *)malloc(sizeof(char) * length);
     for (int i = 0; i < length; i++) {
         gridString[i] = GridController::characterForType(this->grid[0][i]);
```
Comment (gillett-hernandez) [inline Tetris Clone C++/GridController.cpp:163]: here and maybe one other place needs to have the loop variable changed to size_t

**Review by gillett-hernandez — CHANGES_REQUESTED** [review]

**Review by gillett-hernandez — APPROVED** [review]

**Pull request merged**